specification/attestation.adoc

```diff
@@ -1,309 +1,874 @@
+below:
 
-* Create TVM attestation structure in a temporary buffer in per-hart 
-confidential memory
-* Populate TVM TcbEvidenceInfo per the TVM measurements recorded by the TSM
-* Copy additional data from CSR <TBD>
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "Layered Attestation architecture for TVMs"]
+image::layered_attestation.svg[]
+
+It is expected that an implementation will provide implementation-specific
+intrinsics to record measurements of the TSM into the firmware RoT for
+measurement to support the AP-TEE layered RTMs attestation of AP-TEE workloads.
+
+==== Evidence Format
+
+The AP-TEE Attestation Evidence uses the IETF Entity Attestation Token
+(<<EAT>>), formatted as an untagged Concise Binary Object Representation
+(<<CBOR>>) Web Token (<<CWT>>).  An AP-TEE EAT profile is proposed to narrow
+the EAT specification for the AP-TEE use case to enable interoperability.
+
+The CWT is composed of one EAT submodule Claims-Set map where the map values
+are nested tokens for the TVM, TSM and Platform Claims.
+
+The TVM EAT is a CWT tagged CBOR formatted token, wrapped with a
+COSE_Sign1 <<COSE>> envelope. It is signed by the TSM attestation key and must
+contain a TVM provided challenge, e.g. a Relying Party provided nonce for
+establishing Evidence freshness.
+
+The TSM EAT is a CWT tagged CBOR formatted token, wrapped with a
+COSE_Sign1 <<COSE>> envelope. It is signed by the Platform attestation key and
+must contain a hash of the TVM public key, in order to create a cryptographic
+binding between the TVM and the TSM tokens.
+
+The Platform EAT is a CWT tagged CBOR formatted token, wrapped with a
+COSE_Sign1 <<COSE>> envelope. It is signed by the RoT attestation key and
+must contain a hash of the TSM public key, in order to create a cryptographic
+bound between the TSM and the platform tokens.
+
+The AP-TEE layered Evidence structure is represented by the above described
+composition of cryptographically chained EAT tokens. Verifier can then attest
+of an AP-TEE workload trustworthiness by independenly inspecting each token,
+while being able to verify that the TCB represented by one token was used to
+generate the next one.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "AP-TEE Attestation Evidence Format", pdfwidth=65%]
+image::attestation_token.svg[]
+
+The AP-TEE Attestation Evidence is defined by the following Concise Data
+Definition Language (CDDL):
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "AP-TEE Attestation Evidence CDDL"]
+[source,cddl]
+----
+COSE_Sign1 = #6.18(COSE_Sign1_Payload)
+submodule-label = 266 ; EAT Submodule
+
+; Each token is a nested CWT
+platform-token = COSE_Sign1
+tsm-token = COSE_Sign1
+tvm-token = COSE_Sign1
+
+; One EAT Submodule
+riscv-ap-tee-token = (
+   submodule-label => {
+      "platform" => platform-token
+      "tsm" => tsm-token
+      "tvm" => tvm-token
+   }
+)
```

Comment:
I think you want a map here, not a group, right?

```suggestion
riscv-ap-tee-token = {
   submodule-label => {
      "platform" => platform-token
      "tsm" => tsm-token
      "tvm" => tvm-token
   }
}
```

Reply:
Yep, this is a map now.